distributed/tests/test_worker.py

```diff
@@ -55,6 +55,84 @@ def test_health():
         assert 'network-send' in d
 
 
+def test_worker_bad_args(loop):
+    @gen.coroutine
+    def f(c, a, b):
+        aa = rpc(ip=a.ip, port=a.port)
+        bb = rpc(ip=b.ip, port=b.port)
+
+        class NoReprObj(object):
+            def __str__(self):
+                raise ValueError("I have no str representation.")
+            def __repr__(self):
+                raise ValueError("I have no repr representation.")
+
+        response = yield aa.compute(key='x',
+                                    function=dumps(NoReprObj),
+                                    args=dumps(()),
+                                    who_has={})
+        assert not a.active
+        assert response['status'] == 'OK'
+        assert a.data['x']
+        assert c.who_has['x'] == {a.address}
+        assert isinstance(response['compute_start'], float)
+        assert isinstance(response['compute_stop'], float)
+        assert isinstance(response['thread'], Integral)
+
+        def bad_func(x):
+            1 / 0
+
+        response = yield bb.compute(key='y',
+                                    function=dumps(bad_func),
+                                    args=dumps(['x']),
+                                    who_has={'x': [a.address]})
+        assert not b.active
+        assert response['status'] == 'error'
+        assert isinstance(loads(response['exception']), ZeroDivisionError)
+        if sys.version_info[0] >= 3:
+            assert any('1 / 0' in line
+                      for line in pluck(3, traceback.extract_tb(
+                          loads(response['traceback'])))
+                      if line)
+        # TODO: test log item.
```

Comment:
You could either redirect the `logger` using the logging module or else use the `capsys` pytest fixture to capture stderr.  See `distributed/diagnostics/tests/test_progress.py:def check_bar_completed` for an example.